fgpyo/_requirements.py

```diff
@@ -0,0 +1,24 @@
+"""Enforce requirements."""
+
+from typing import Optional
+
+
+class RequirementError(ValueError):
+    """Exception raised when a requirement is not satisfied."""
```

Comment:
I can think of some requirements that wouldn't necessary relate to values.

Wondering if it'd be better to inherit from `AssertionError` instead?

But since you are not using `assert` below (probably to avoid someone turning off the behavior in Python's debug mode), maybe inheriting from the `Exception` class is ideal.

Reply:
And the standard library docs explicitly instructs developers to inherit from `Exception` when defining their own exceptions. (emphasis mine)

> All built-in, non-system-exiting exceptions are derived from this class. **All user-defined exceptions should also be derived from this class.**

https://docs.python.org/3/library/exceptions.html#Exception

TIL!